integration/docker/Dockerfile

```diff
@@ -27,4 +27,7 @@ RUN cd /opt && \
 COPY conf /opt/alluxio/conf/
 COPY entrypoint.sh /
 
+RUN adduser -u 1000 -S alluxio -G root
```

Comment:
nit: can we add default env variables so that we can override at runtime if needed?
```
ENV ALLUXIO_USER=alluxio
ENV ALLUXIO_USER_ID=1000
ENV ALLUXIO_GROUP=root
```

Reply:
The non-root user is created on Docker Build, I can make them args instead so they can be easily overridden as build arguments.

I've noticed that this breaks when pulling the `1.8.1` tar, but works when pulling the `2.0.0-RC3` tar.

```bash
# 1.8.1
bash-4.4$ ls -la
total 64
drwxr-xr-x    1 501      dialout       4096 Jun 14 07:15 .
drwxr-xr-x    1 501      dialout       4096 Jun 14 07:15 ..
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._alluxio-env.sh.template
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._alluxio-site.properties.template
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._core-site.xml.template
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._log4j.properties
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._masters
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._metrics.properties.template
-rw-r--r--    1 501      dialout        239 Sep 27  2018 ._workers
lrwxrwxrwx    1 root     root            37 Jun 13 09:16 alluxio-env.sh.template -> ../../../conf/alluxio-env.sh.template
lrwxrwxrwx    1 root     root            46 Jun 13 09:16 alluxio-site.properties.template -> ../../../conf/alluxio-site.properties.template
-rw-r--r--    1 501      dialout       3737 Sep 27  2018 core-site.xml.template
-rw-r--r--    1 501      dialout       7157 Sep 27  2018 log4j.properties
-rw-r--r--    1 501      dialout       1058 Sep 27  2018 masters
-rw-r--r--    1 501      dialout       3127 Sep 27  2018 metrics.properties.template
-rw-r--r--    1 501      dialout        587 Sep 27  2018 workers
```
```bash
# 2.0.0-RC3
bash-4.4$ ls -la
total 80
drwxr-xr-x    1 alluxio  1000          4096 Jun 14 07:25 .
drwxr-xr-x    1 root     root          4096 Jun 14 07:24 ..
-rw-r--r--    1 alluxio  1000         27040 Jun  6 06:31 LICENSE
drwxr-xr-x    2 alluxio  1000          4096 Jun 14 07:24 assembly
drwxr-xr-x    2 alluxio  1000          4096 Jun 14 07:24 bin
drwxr-xr-x    2 alluxio  1000          4096 Jun 14 07:24 client
drwxr-xr-x    1 alluxio  1000          4096 Jun 14 07:25 conf
drwxr-xr-x    8 alluxio  1000          4096 Jun 14 07:24 integration
drwxr-xr-x    2 alluxio  1000          4096 Jun 14 07:24 lib
drwxr-xr-x    2 alluxio  1000          4096 Jun 14 07:24 libexec
drwxr-xr-x    3 alluxio  1000          4096 Jun 14 07:24 logs
-rw-r--r--    1 alluxio  root             0 Jun 14 07:25 test
drwxr-xr-x    2 alluxio  1000          4096 Jun  6 06:31 underFSStorage
drwxr-xr-x    4 alluxio  1000          4096 Jun 14 07:24 webui
```

Reply:
+1 creating at build time makes sense if we want to launch container with that user :-)

Reply:
@rblaine95 `ADD` and `COPY` directives do not honor `USER` in the `Dockerfile`.  adding the following to the end, makes sure the permissions are setup properly
```
RUN adduser -u 1000 -S alluxio -G root
RUN chown alluxio:root /entrypoint.sh
RUN chown alluxio:root -R /opt/alluxio*

USER 1000
```

Reply:
if would parametrize this with build arg and run only if arg != root